kms.tf

```diff
@@ -1,20 +1,20 @@
-data "aws_caller_identity" "current" {
-}
+data "aws_caller_identity" "current" {}
 
 resource "aws_kms_alias" "kms" {
   name          = "alias/secrets"
-  target_key_id = "${aws_kms_key.main.key_id}"
+  target_key_id = aws_kms_key.main.key_id
 }
 
 # KMS requires that the creator has access to the key so you don't lock yourself out
 locals {
-  my_role_name = "${split("/", data.aws_caller_identity.current.arn)[1]}"
-  my_role_arn  = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.my_role_name}"
+  current_arn        = data.aws_caller_identity.current.arn
+  current_account_id = data.aws_caller_identity.current.account_id
+  my_role_arn        = length(regexall("/", local.current_arn)) > 0 ? "arn:aws:iam::${local.current_account_id}:role/${split("/", local.current_arn)[1]}" : local.current_arn
```

Comment:
@aarongorka , I finally got back to this. Do you think this will work? It introduces a dependency on 0.12.7 which I think is ok, and that length/regexall is Terraform craziness for, "does this string contain regex "/"" ? So basically, if the current_arn contains a forward slash, use your expression ; otherwise, just use the current_arn and assume we logged in by keys etc.